sklearn/model_selection/tests/test_validation.py

```diff
@@ -270,16 +271,17 @@ def get_params(self, deep=False):
 # XXX: use 2D array, since 1D X is being detected as a single sample in
 # check_consistent_length
 X = np.ones((10, 2))
-X_sparse = coo_matrix(X)
 y = np.array([0, 0, 1, 1, 2, 2, 3, 3, 4, 4])
 # The number of samples per class needs to be > n_splits,
 # for StratifiedKFold(n_splits=3)
 y2 = np.array([1, 1, 1, 2, 2, 2, 3, 3, 3, 3])
-P_sparse = coo_matrix(np.eye(5))
+P_sparse = COO_CONTAINERS[-1](np.eye(5))
```

Comment:
It is not what I meant. What I meant was to change the usage of `P_sparse` and use `P` in the Mock classifier since we are only interested about the shape of the data. The sparsity in the Mock does not matter so much.